### PR TITLE
Compatibility/java interop

### PR DESCRIPTION
**Pull request by SHA-4096**
# What's the purpose of the change
## 1. handler mismatch
When interoperating with dubbo-java, there may be an issue when client from java side try to invoke an rpc function whose first-letter is in lower case, since dubbo-go now only provide handler whose method name is always with upper-cased first letter, due to the language feature of golang. It would cause failure in rpc function invocation, as mentioned in https://github.com/apache/dubbo-go-samples/pull/771 .

To tackle this problem, a straightforward idea is to register another handler with a different first-letter case, and that's the change made by this pull request.

## 2. content-type codec mismatch
When using hessian2 for serialization, some version of java-side client would set a wrong content-type in request, causing marshal and unmarshal error. In order to tackle the problem, a backupCodec is introduced to handle the message in the server-side serialization type. 

# Validity of the solution
## solution for problem 1
After adding a breakpoint at `dubbo-go/server/server.go` at line 254, which is the section that exports service , we can see the registered handlers on `service->serviceOpts->rpsService->s->triServer->handlers`,  and  `/org.apache.dubbo.tri.hessian2.api.GreetingsService/greet` is made possible by changes in this PR. 
image

## solution for problem 2
A warn message would be logged when request content-type and server serialization mismatches, and the server would try to marshal/unmarshal message in server serialization codec. 
image

### CODE REVIEW

**Comment by chickenlj** [comment]
Please run `make fmt` to format the code

File: common/dubboutil/case_invert.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package dubboutil
```
Comment (chickenlj) [inline common/dubboutil/case_invert.go:1]: Add apache license header for this file

File: common/rpc_service.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package common
 
 import (
 	"context"
+	"dubbo.apache.org/dubbo-go/v3/common/dubboutil"
```
Comment (FoghostCn) [inline common/rpc_service.go:22]: pls format imports

File: server/server.go [hunk 3]
```diff
@@ -180,6 +181,18 @@ func (s *Server) exportServices() (err error) {
 			err = svcOpts.ExportWithoutInfo()
 		} else {
 			info := infoRaw.(*ServiceInfo)
+			//Add a method with a name of a differtent first-letter case
+			//to achieve interoperability with java
+			var additionalMethods []MethodInfo
+			for _, method := range info.Methods {
+				newMethod := method
+				newMethod.Name = dubboutil.SwapCaseFirstRune(method.Name)
```
Comment (chickenlj) [inline server/server.go:189]: Do we need a deep copy of `method`?

Comment (SHA-4096) [reply server/server.go:189]: I'll have a look at other solutions

File: common/dubboutil/case_invert.go (hunk 1 above)
Comment (SHA-4096) [reply common/dubboutil/case_invert.go:20]: Added

File: common/rpc_service.go (hunk 2 above)
Comment (SHA-4096) [reply common/rpc_service.go]: formatted

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**